rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -1606,8 +1655,16 @@ private predicate useImportEdge(Use use, string name, ItemNode item, SuccessorKi
         not tree.hasRename() and
         name = item.getName()
         or
-        name = tree.getRename().getName().getText() and
-        name != "_"
+        exists(Rename rename | rename = tree.getRename() |
+          name = rename.getName().getText()
+          or
+          // When the rename doesn't have a name it's an underscore import. This
+          // makes the imported item visible but unnameable. We represent this
+          // by using the name `_` which can never occur in a path.  See also:
+          // https://doc.rust-lang.org/reference/items/use-declarations.html#r-items.use.as-underscore
+          not rename.hasName() and
+          name = "_"
+        )
```

Comment:
Note, the old disjunct here did in fact not do anything. When there is a rename it's name is never `_`. That is,
```
predicate isUnderscope(UseTree tree) { tree.getRename().getName().getText() = "_" }
```
is empty. Instead `not rename.hasName()` holds when `_` is used in the source code.